pkg/streamingpromql/types/hpoint_ring_buffer.go

```diff
@@ -148,9 +174,22 @@ func (b *HPointRingBuffer) Append(p promql.HPoint) error {
 	}
 
 	nextIndex := (b.firstIndex + b.size) % len(b.points)
-	b.points[nextIndex] = p
 	b.size++
-	return nil
+	return &b.points[nextIndex], nil
+}
+
+func (b *HPointRingBuffer) RemoveLastPoint() {
+	if b.size == 0 {
+		return
```

Comment:
I'm tempted to panic here - if we're calling this method but there are no points, that sounds like a bug.